src/Oauth1.php

```diff
@@ -239,8 +239,11 @@ private function prepareParameters($data)
      */
     private function signUsingHmacSha1($baseString)
     {
-        $key = rawurlencode($this->config['consumer_secret'])
-            . '&' . rawurlencode($this->config['token_secret']);
+        $key = rawurlencode($this->config['consumer_secret']);
+
+        if (!empty($this->config['token_secret'])) {
+            $key .= '&' . rawurlencode($this->config['token_secret']);
```

Comment:
I made this change at https://github.com/guzzle/oauth-subscriber/pull/49 and I can confirm that it works. Otherwise, the signatures won't match.